sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlockBlobClient.java

```diff
@@ -137,10 +138,28 @@ public BlobOutputStream getBlobOutputStream(BlobRequestConditions requestConditi
     public BlobOutputStream getBlobOutputStream(ParallelTransferOptions parallelTransferOptions,
         BlobHttpHeaders headers, Map<String, String> metadata, AccessTier tier,
         BlobRequestConditions requestConditions) {
-        return BlobOutputStream.blockBlobOutputStream(client, parallelTransferOptions, headers, metadata, tier,
+
+        BlobAsyncClient blobClient = prepareBuilder(client).buildAsyncClient();
+
+        return BlobOutputStream.blockBlobOutputStream(blobClient, parallelTransferOptions, headers, metadata, tier,
             requestConditions);
     }
 
+    private BlobClientBuilder prepareBuilder(BlobAsyncClientBase client) {
```

Comment:
Could make this static as it access no instance level properties, on the flip side could remove the parameter and access all the values in the instance.

Reply:
Just removed the param